File: .puku-cli/validate-workflows.js
#!/usr/bin/env node
// Validate YAML syntax of GitHub workflow files.
const fs = require('fs');
const yaml = require('js-yaml');

const files = [
  '.github/workflows/project-sync.yml',
  '.github/workflows/milestone-sync.yml',
  '.github/workflows/nextjs.yml',
  '.github/dependabot.yml',
  '.github/stale.yml',
  '.github/ISSUE_TEMPLATE/bug.yml',
  '.github/ISSUE_TEMPLATE/feature.yml',
  '.github/ISSUE_TEMPLATE/content.yml',
  '.github/ISSUE_TEMPLATE/rfc.yml',
  '.github/ISSUE_TEMPLATE/config.yml',
];

let ok = true;
for (const f of files) {
  try {
    const text = fs.readFileSync(f, 'utf8');
    const data = yaml.load(text);
    const size = yaml.dump(data).length;
    const name = f.split('/').pop().padEnd(40);
    console.log(`OK  ${name} (${size} chars)`);
  } catch (e) {
    console.error(`ERR ${f}: ${e.message}`);
    ok = false;
  }
}

process.exit(ok ? 0 : 1);

File: .puku-cli/validate-workflows.py
#!/usr/bin/env python3
"""Validate YAML syntax of GitHub workflow files."""
import sys
import yaml

files = [
    "/Users/raka/github.com/raihankhan/portfolio/.github/workflows/project-sync.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/workflows/milestone-sync.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/workflows/nextjs.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/dependabot.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/stale.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/ISSUE_TEMPLATE/bug.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/ISSUE_TEMPLATE/feature.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/ISSUE_TEMPLATE/content.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/ISSUE_TEMPLATE/rfc.yml",
    "/Users/raka/github.com/raihankhan/portfolio/.github/ISSUE_TEMPLATE/config.yml",
]

ok = True
for f in files:
    try:
        with open(f) as fh:
            data = yaml.safe_load(fh)
        size = len(yaml.safe_dump(data))
        print(f"OK  {f.split('/')[-1]:40s} ({size} chars)")
    except yaml.YAMLError as e:
        print(f"ERR {f}: {e}")
        ok = False
    except FileNotFoundError:
        print(f"??  {f}: not found")

sys.exit(0 if ok else 1)
